src/lib.rs

```diff
@@ -130,3 +130,13 @@ pub use self::keys::{
     keyval_name,
     keyval_to_unicode
 };
+
+pub fn pango_layout_get_clip_region(layout: &pango::Layout, x_origin: i32, y_origin: i32, index_ranges: &[i32]) -> Option<cairo::Region> {
+    use glib::translate::from_glib_full;
+    use glib::translate::ToGlibPtr;
+
+    assert_initialized_main_thread!();
```

Comment:
Assert need be on first line.
Also IMHO better add separate "functions.rs" now.

Reply:
Ah you manually implemented it. See my comments here https://github.com/gtk-rs/gdk/pull/224/files#r199389013

Reply:
You probably want the array to be an array of tuples

Reply:
I don't know how tuples are represented in memory... If it's contiguous values, then it'll easy but otherwise it'll require a copy, not sure we want that.

Reply:
Currently it would work but there is no guarantee I believe. The problem otherwise is that you have an easy to misuse API

Reply:
Maybe a tuple struct with packed representation would be best here. Ugly ugly C API

Reply:
Talk about dirty hacks...